src/sentry/uptime/endpoints/organization_uptime_stats.py

```diff
@@ -0,0 +1,175 @@
+import logging
+from collections import defaultdict
+
+from drf_spectacular.utils import extend_schema
+from google.protobuf.timestamp_pb2 import Timestamp
+from rest_framework.request import Request
+from rest_framework.response import Response
+from sentry_protos.snuba.v1.endpoint_time_series_pb2 import TimeSeriesRequest, TimeSeriesResponse
+from sentry_protos.snuba.v1.request_common_pb2 import RequestMeta, TraceItemType
+from sentry_protos.snuba.v1.trace_item_attribute_pb2 import (
+    AttributeAggregation,
+    AttributeKey,
+    AttributeValue,
+    Function,
+    StrArray,
+)
+from sentry_protos.snuba.v1.trace_item_filter_pb2 import ComparisonFilter, TraceItemFilter
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import StatsArgsDict, StatsMixin, region_silo_endpoint
+from sentry.api.bases.organization import OrganizationEndpoint, OrganizationPermission
+from sentry.models.organization import Organization
+from sentry.models.project import Project
+from sentry.uptime.models import ProjectUptimeSubscription, UptimeSubscription
+from sentry.utils.snuba_rpc import timeseries_rpc
+
+logger = logging.getLogger(__name__)
+
+
+@region_silo_endpoint
+@extend_schema(tags=["Crons"])
+class OrganizationUptimeStatsEndpoint(OrganizationEndpoint, StatsMixin):
+    publish_status = {
+        "GET": ApiPublishStatus.EXPERIMENTAL,
+    }
+    owner = ApiOwner.CRONS
+    permission_classes = (OrganizationPermission,)
+
+    def get(self, request: Request, organization: Organization) -> Response:
+        timerange_args = self._parse_args(request)
+        projects = self.get_projects(request, organization)
+
+        uptime_subscription_ids = request.GET.getlist("uptime_subscription_id")
+
+        if not uptime_subscription_ids:
+            return self.respond("No uptime subscription ids provided", status=400)
+
+        try:
+            self._validate_uptime_subscription_ids(uptime_subscription_ids, projects)
+        except ValueError:
+            return self.respond("Invalid uptime subscription ids provided", status=400)
+
+        try:
+            responses = self._make_eap_request(
+                organization, projects, uptime_subscription_ids, timerange_args
+            )
+        except Exception:
+            logger.exception("Error making EAP RPC request for uptime check stats")
```

Comment:
nit: Typically I think we use machine-readable style names such as 
```suggestion
            logger.exception("uptime.eap_stats_request_error")
```

Reply:
are you sure? looks to be mostly human readable from what i've seen. regardless, this will report the error to sentry, which is the primary reason i'm doing `logger.exception`

Reply:
oh yeah maybe you're right. idk